### DISCUSSION TITLE
Support for Ring Intercom

### DISCUSSION THREAD

**Discussion by simps100** [tsightler/ring-mqtt#479] [Feature Requests]
Hey

Have pre ordered a ring intercom which looks like it can take control of my flat intercom and make it smart - will there be any plans to support this? https://amzn.eu/d/cPccLjL

Happy to help in any testing if needed when mine arrives hopefully the end of the month!

Thanks
Marc

**Comment by tsightler** [comment]
No plans.  Doesn't appear that there are plans for support in the underlying [ring-client-api ](https://github.com/dgreif/ring/issues/1062) either.  I'll never own one of these so the chances of it being implemented by me is basically zero.  If someone out there does figure out the API and something useful to do with it, and submits a PR, that's about the only way it would ever be supported.

**Comment by gforums** [comment]
Dear all, I saw that the ring client api is now supporting the intercom device. Might that raise chances that this one will also support it? Thanks a lot. Best wishes, Christian

**Reply by tsightler** [reply]
As state above, I don't own an intercom, and I have zero reason to own one, so the only way it is likely to be supported is if someone is either a) willing to share their device with me via the Ring app, or b) does the work and submits a PR.

I'm not opposed to doing the work now that the device is supported in ring-client-api it's much more doable, however, it's different enough from other devices that I'm not sure I could code up support without having access to the device.

The other thing that would be needed is someone to define what they would actually want to do with it.  Audio would not be supported, and the only functions in ring-client-api or basically a "doorbell", to detect when the button is pressed, and a "lock" which can be used to unlock the door and then some basic data like battery monitoring.  That's all that could be supported in ring-mqtt as well.

If someone steps up and is willing to share access to their device for a short while, I'm happy to take a look at it, until then, support is unlikely.

**Reply by gforums** [reply]
Thanks a lot, I'd be willing to share the device. I think mainly the unlock functionality would be very helpful. There's no direct message function here I assume? If you drop me a short mail at christian@gforums.de, I'd prepare to share the intercom.
Thanks again for your offer!

**Reply by tsightler** [reply]
OK, it may be a few days before I'm ready.  I've been reviewing the ring-homebridge code to use as a reference and I'll try to code up at first attempt before requesting access because I don't want to keep access longer than a minimum amount of time as I know it's quite an intrusion..

**Reply by gforums** [reply]
Alright, just get back to me as soon as you'd like give it a try. Have a great evening, thanks.

**Reply by simps100** [reply]
Doorbell and Unlock functions would fantastic in Homeassistant for me also - thank you for looking into supporting this willing to test etc with mine also if it helps.

**Reply by tsightler** [reply]
Hey all,

After studying the implementation from homebridge-ring I decided to give it a spin just coding it up without access.  Could one of you test out the dev branch and tell me if it finds your intercom and publishes the entities?  It should publish a device with 4 entities, the doorbell is a simple binary sensor in HA, the unlock function should appear as a lock in HA, and there should also be an info sensor (without much actual data) and a battery entity as well.

The way it should work is that any unlock, either via Home Assistant, or Ring app, should trigger the lock and should display in HA as unlocked for 5 seconds.  For a ding, the binary sensor should show "ON" for 15 seconds before returning to off.

The chances of me getting all of this correct on the first try isn't super high, but if you'd be willing to give it a shot and report back.  Logs from any issues would also be super useful.  You can send logs to my email at the same username at Gmail.  Thanks for the help!

**Reply by gforums** [reply]
Wow, great to hear. I can't promise I'll be able to try before the weekend but as soon as I did, I'll let you know!

**Comment by gforums** [comment]
Okay, did some first testing and looks good.
But as I am new to mqtt, give me some more time please :)

Was able to subscribe to 'ring/#' and found my devices.
I can confirm that I can receive ring/..../intercom/.../lock and info and ding states.

Shows me state = LOCKED and a correct battery level.

Trying more tomorrow (especially if unlock works).

For the first approach, it looks more than promising. Great!
Thanks.

**Comment by gforums** [comment]
Update: AMAZING!

I found the three channels from the intercom within openhab mqtt broker automatically.
The manual installation went very well.

It works to unlock the door via mqtt.
It works to get a.command when doorbell is pressed.
Status information also works.

@tsightler: I am very impressed and thankful. I think this can be moved to the main branch.

Best,
Christian

![IMG_20230218_163050](https://user-images.githubusercontent.com/82218762/219874313-f7f3d7a0-4524-41ab-a22d-1fcd491db2d4.jpg)

**Reply by tsightler** [reply]
Thanks for the testing and positive feedback.  It will be included in the 5.2.0 release, which will probably be in a few weeks.  I'm waiting on some fixes to make their way into some upstream libraries and also one bug that I'm still digging into around streaming, plus I'm going to be out of town for most of the next 3 weeks, and I don't like pushing out updates in that case since, if I somehow missed something significant, I wouldn't be around to fix it.

**Reply by Corezone** [reply]
i have testet it also :) works fine thanks a lot :)

**Reply by tsightler** [reply]
Intercom support is no released in 5.2.0.

**Reply by isibizi** [reply]
Thx for this update, but the add-on can't find my intercom

![image](https://user-images.githubusercontent.com/73044043/224928005-ebe247b6-edeb-47c1-9f44-5ccf528fd1a3.png)

**Reply by isibizi** [reply]
OK its working now, "enable_cameras" must be activated
